src/reblox/grid.rs

```diff
@@ -0,0 +1,595 @@
+    match rotation {
+        0 =>
+            match block_type {
+                BlockType::I => [pos - grid_width, pos, pos + grid_width, pos - grid_width * 2],
+                BlockType::J => [pos - grid_width, pos, pos + grid_width, pos - grid_width - 1],
+                BlockType::L => [pos - grid_width, pos, pos + grid_width, pos - grid_width + 1],
+                BlockType::O => [pos, pos + 1, pos - grid_width, pos - grid_width + 1],
+                BlockType::S => [pos, pos + 1, pos - grid_width, pos - grid_width - 1],
+                BlockType::T => [pos - 1, pos, pos + 1, pos - grid_width],
+                BlockType::Z => [pos - 1, pos, pos - grid_width, pos - grid_width + 1],
+                _ => [0, 0, 0, 0],
+            },
+        1 =>
+            match block_type {
+                BlockType::I => [pos + 1, pos, pos - 1, pos - 2],
+                BlockType::J => [pos + 1, pos, pos - 1, pos + grid_width - 1],
+                BlockType::L => [pos + 1, pos, pos - 1, pos - grid_width - 1],
+                BlockType::O => [pos, pos - 1, pos - grid_width, pos - grid_width - 1],
+                BlockType::S => [pos - grid_width, pos, pos - 1, pos + grid_width - 1],
+                BlockType::T => [pos + grid_width, pos, pos - grid_width, pos - 1],
+                BlockType::Z => [pos + grid_width, pos, pos - 1, pos - grid_width - 1],
+                _ => [0, 0, 0, 0],
+            },
+        2 =>
+            match block_type {
+                BlockType::I => [pos - grid_width, pos, pos + grid_width, pos - grid_width * 2],
+                BlockType::J => [pos - grid_width, pos, pos + grid_width, pos + grid_width + 1],
+                BlockType::L => [pos - grid_width, pos, pos + grid_width, pos + grid_width - 1],
+                BlockType::O => [pos, pos - 1, pos + grid_width, pos + grid_width - 1],
+                BlockType::S => [pos - 1, pos, pos + grid_width, pos + grid_width + 1],
+                BlockType::T => [pos - 1, pos, pos + 1, pos + grid_width],
+                BlockType::Z => [pos + 1, pos, pos + grid_width, pos + grid_width - 1],
+                _ => [0, 0, 0, 0],
+            },
+        3 =>
+            match block_type {
+                BlockType::I => [pos + 1, pos, pos - 1, pos - 2],
+                BlockType::J => [pos - 1, pos, pos + 1, pos - grid_width + 1],
+                BlockType::L => [pos - 1, pos, pos + 1, pos + grid_width + 1],
+                BlockType::O => [pos, pos + 1, pos + grid_width, pos + grid_width + 1],
+                BlockType::S => [pos + grid_width, pos, pos + 1, pos - grid_width + 1],
+                BlockType::T => [pos + 1, pos + grid_width, pos, pos - grid_width],
+                BlockType::Z => [pos - grid_width, pos, pos + 1, pos + grid_width + 1],
+                _ => [0, 0, 0, 0],
+            },
+        _ => panic!("fn pr2piece: Invalid rotation value!"),
+    }
+}
+
+/// (0, 0) is bottom left
```

Comment:
You should add proper docs to this struct and its field: some of it might not be obvious at first.